features/step_definitions/visitor_can_register_for_an_account.feature

```diff
@@ -0,0 +1,40 @@
+Feature: Visitor can register for an account
+
+    As a Visitor,
+    In order to access unique features on the news app,
+    I would like to be able to register for an account on the site.
+
+    Background:
+        Given the following user exists
+        | name | email          | password | password_confirmation |
+        | Bill | bill@email.com | password | password              |
+        Given I visit the landing page
+        And I click 'Register'
+
+    Scenario: When I fill in the sign up form correctly, I can sign up
+        Given I fill in 'Email' field with 'mail@email.com'
+        And I fill in 'Password' field with 'password'
+        And I fill in 'Password confirmation' field with 'password'
+        And I click 'Sign up'
+        Then I should see 'Welcome! You have signed up successfully.'
+
+    Scenario: When password is too short, I get an error message
```

Comment:
You can have in every sad path in one scenario, similar to this example:
```
Scenario: When I don't fill in the sign up form I get error messages
        When I click 'Sign up'
        Then I should see "Email can't be blank"
        And I should see "Password can't be blank"
        And I should see "Name can't be blank"
        And I should see "Address can't be blank"
        And I should see "Phone can't be blank"

 Scenario: When Password and Password confirmation doesn't match I get error message
        When I fill in 'Password' field with 'password'
        And I fill in 'Password confirmation' field with 'pasword'
        And I click 'Sign up'
        Then I should see "Password confirmation doesn't match Password"

    Scenario: When Email is taken I get error message
        When I fill in 'Email' field with 'real@email.com'
        And I click 'Sign up'
        Then I should see "Email has already been taken"
```

Reply:
Done